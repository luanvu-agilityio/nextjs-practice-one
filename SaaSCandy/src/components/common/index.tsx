export * from './Heading';
export * from './Typography';
export * from './Button';
export * from './IconButton';
export * from './IconButton';
export * from './InputController';
export * from './Toast';
export * from './Breadcrumb';
export * from './Divider';
export * from './Section';
export * from './CheckboxController';
export * from './Skeleton';
export * from './DropdownMenu';
export * from './Dialog';
export * from './Spinner';
