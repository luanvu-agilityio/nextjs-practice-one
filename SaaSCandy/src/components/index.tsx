export * from './common';
export * from './layout';
export * from './form';
export * from './pages';
export * from './NavBar';
export * from './SocialButton';
export * from './UserMenu';
export * from './FeaturedCard';
export * from './Share';
export * from './NewsletterSignup';
export * from './Tags';
export * from './BlogCard';
export * from './BlogList';
export * from './ContentRenderer';
export * from './PricingPlanCard';
export * from './Switch';
export * from './FAQSection';
export * from './HeroSection';
export * from './JoinUsSection';
