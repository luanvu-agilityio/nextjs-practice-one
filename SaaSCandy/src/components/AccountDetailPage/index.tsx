'use client';

import { Fragment, useEffect } from 'react';
import Link from 'next/link';
import { useSession, signOut } from 'next-auth/react';
import { useRouter } from 'next/navigation';

// Components
import { Button, Heading, Typography } from '@/components/common';
import {
  Breadcrumb,
  BreadcrumbList,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbSeparator,
  BreadcrumbPage,
} from '@/components/common/Breadcrumb';
import { Spinner } from '../ui/spinner';

// Icons
import LogoIcon from '@/components/icons/Logo';

// Utils & Constants
import { ROUTES } from '@/constants';

function AccountDetailPage() {
  const { data: session, status } = useSession();
  const router = useRouter();

  // Redirect if not authenticated
  useEffect(() => {
    if (status === 'unauthenticated') {
      router.push(ROUTES.SIGN_IN);
    }
  }, [status, router]);

  if (status === 'loading') {
    return (
      <div className='flex justify-center items-center min-h-screen'>
        <Spinner />
      </div>
    );
  }

  if (!session?.user) {
    return null;
  }

  const {
    user: { name, email, image },
  } = session;

  const handleGoBack = () => {
    router.back();
  };

  const handleSignOut = async () => {
    try {
      await signOut({
        callbackUrl: ROUTES.HOME,
      });
    } catch (error) {
      console.error('Sign out error:', error);
    }
  };

  const breadcrumbs = [
    { label: 'Home', href: '/', isActive: false },
    { label: 'Account', href: '/account', isActive: true },
  ];

  return (
    <div className='flex flex-col items-center max-w-[1296px] mx-auto min-h-screen py-8'>
      {/* Header Section */}
      <div className='pb-12 w-full'>
        <Heading
          as='h1'
          size='xl'
          content='Account Details'
          className='text-center'
        />

        {/* Breadcrumb */}
        <Breadcrumb className='justify-center mt-4'>
          <BreadcrumbList>
            {breadcrumbs.map((item, index) => (
              <Fragment key={item.label}>
                <BreadcrumbItem>
                  {item.isActive ? (
                    <BreadcrumbPage>{item.label}</BreadcrumbPage>
                  ) : (
                    <BreadcrumbLink href={item.href}>
                      {item.label}
                    </BreadcrumbLink>
                  )}
                </BreadcrumbItem>
                {index < breadcrumbs.length - 1 && <BreadcrumbSeparator />}
              </Fragment>
            ))}
          </BreadcrumbList>
        </Breadcrumb>
      </div>

      {/* Account Detail Section */}
      <div className='py-14 px-16 w-full max-w-159 rounded-4xl border border-form-border-color shadow-form bg-white'>
        <div className='flex flex-col gap-10'>
          {/* Logo */}
          <div className='flex justify-center'>
            <Link href={ROUTES.HOME} className='flex items-center gap-3'>
              <LogoIcon className='w-10 h-10' />
              <span className='text-3xl font-secondary text-primary'>
                SaaS<span className='font-medium'>Candy</span>
              </span>
            </Link>
          </div>

          {/* User Information */}
          <div className='space-y-8'>
            {/* Profile Header */}
            <div className='text-center space-y-4'>
              {image ? (
                <div className='flex justify-center'>
                  <Link
                    href={image}
                    aria-label='Profile picture'
                    className='w-24 h-24 rounded-full border-4 border-orange-background'
                  />
                </div>
              ) : (
                <div className='flex justify-center'>
                  <div className='w-24 h-24 rounded-full bg-orange-background flex items-center justify-center text-white text-2xl font-bold'>
                    {name?.charAt(0) || email?.charAt(0) || 'U'}
                  </div>
                </div>
              )}

              <div>
                <Heading
                  as='h2'
                  size='md'
                  className='text-primary'
                  content={name || 'User'}
                />

                <Typography className='text-gray-background mt-1'>
                  {email}
                </Typography>
              </div>
            </div>

            {/* Account Information Cards */}
            <div className='space-y-4'>
              {/* Personal Information */}
              <div className='bg-blue-foreground opacity-80 rounded-lg p-6'>
                <Heading
                  as='h3'
                  size='md'
                  className='text-primary mb-4'
                  content='Personal Information'
                />

                <div className='space-y-3'>
                  <div className='flex justify-between items-center'>
                    <span className='text-blue-background font-bold'>
                      Name:
                    </span>
                    <span className='font-medium text-primary'>
                      {name || 'Not provided'}
                    </span>
                  </div>
                  <div className='flex justify-between items-center'>
                    <span className='text-blue-background font-bold'>
                      Email:
                    </span>
                    <span className='font-medium text-primary'>{email}</span>
                  </div>
                  <div className='flex justify-between items-center'>
                    <span className='text-blue-background font-bold'>
                      Account Type:
                    </span>
                    <span className='px-2 py-1 bg-blue-foreground text-blue-background rounded-full text-md'>
                      {image ? 'Social Account' : 'Email Account'}
                    </span>
                  </div>
                </div>
              </div>

              {/* Account Settings */}
              <div className='bg-blue-foreground opacity-80 rounded-lg p-6'>
                <Heading
                  as='h3'
                  size='md'
                  className='text-primary mb-4'
                  content='Account Settings'
                />

                <div className='space-y-3'>
                  <Button
                    variant='secondary'
                    className='w-full justify-start font-semibold'
                    disabled
                  >
                    Edit Profile (Coming Soon)
                  </Button>
                  <Button
                    variant='secondary'
                    className='w-full justify-start font-semibold'
                    disabled
                  >
                    Change Password (Coming Soon)
                  </Button>
                  <Button
                    variant='secondary'
                    className='w-full justify-start font-semibold'
                    disabled
                  >
                    Privacy Settings (Coming Soon)
                  </Button>
                </div>
              </div>
            </div>

            {/* Action Buttons */}
            <div className='flex gap-4 pt-6'>
              <Button
                variant='secondary'
                className='flex-1'
                onClick={handleGoBack}
              >
                Back
              </Button>
              <Button
                variant='tertiary'
                className='flex-1'
                onClick={handleSignOut}
              >
                Sign Out
              </Button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default AccountDetailPage;
