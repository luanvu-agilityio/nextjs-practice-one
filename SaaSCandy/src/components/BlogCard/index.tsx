'use client';

import React from 'react';
import Image from 'next/image';
import Link from 'next/link';

// Components
import { Heading } from '@/components/common';

interface BlogCardProps {
  slug: string;
  title: string;
  image?: string;
  date: string;
}

function BlogCard({ slug, title, image, date }: Readonly<BlogCardProps>) {
  return (
    <article className='group'>
      <Link
        href={`/blog/${slug}`}
        className='block rounded-lg overflow-hidden hover:shadow-md transition-shadow bg-white'
      >
        <div className='w-full overflow-hidden'>
          {image ? (
            <Image
              alt={title}
              className='object-cover w-full h-full group-hover:scale-105 transition-transform duration-300'
              src={image}
              width={416}
              height={284}
            />
          ) : (
            <div className='w-full h-full flex items-center justify-center text-gray-background'>
              No image
            </div>
          )}
        </div>

        <div className='flex flex-col gap-4 py-10'>
          <Heading
            as='h3'
            size='md'
            content={title}
            className='text-primary font-medium text-xl'
          />
          <div className='text-xs text-gray-background'>{date}</div>
        </div>
      </Link>
    </article>
  );
}
export { BlogCard };
