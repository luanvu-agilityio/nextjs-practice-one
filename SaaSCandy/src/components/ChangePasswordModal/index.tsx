'use client';

import { useState } from 'react';
import { useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';
import { useSession } from '@/lib/auth-client';

// Components
import { Button, showToast } from '@/components/common';
import { ErrorMessage } from '../common/ErrorMessage';
import { InputController } from '@/components/common/InputController';
import {
  Dialog,
  DialogContent,
  DialogHeader,
  DialogTitle,
  DialogFooter,
} from '@/components';

// Utils
import { ChangePasswordFormData, changePasswordSchema } from '@/utils';

// Types
import { TOAST_VARIANTS } from '@/types';

// Service
import { changePassword } from '@/service';

interface ChangePasswordModalProps {
  open: boolean;
  onOpenChange: (open: boolean) => void;
  onSuccess: () => void;
}

function ChangePasswordModal({
  open,
  onOpenChange,
  onSuccess,
}: Readonly<ChangePasswordModalProps>) {
  const { data: session } = useSession();
  const [isSubmitting, setIsSubmitting] = useState(false);
  const [error, setError] = useState<string | null>(null);

  const user = session?.user;

  const { control, handleSubmit, reset } = useForm<ChangePasswordFormData>({
    resolver: zodResolver(changePasswordSchema),
    mode: 'onBlur',
    defaultValues: {
      currentPassword: '',
      newPassword: '',
      confirmPassword: '',
    },
  });

  const onSubmit = async (data: ChangePasswordFormData) => {
    if (!user?.id) {
      setError('User session not found');
      return;
    }

    setIsSubmitting(true);
    setError(null);

    try {
      const result = await changePassword(
        data.currentPassword,
        data.newPassword
      );

      if (!result.success) {
        throw new Error(result.error || 'Failed to update password');
      }

      showToast({
        title: 'Password Updated',
        description: 'Your password has been changed successfully.',
        variant: TOAST_VARIANTS.SUCCESS,
        duration: 3000,
      });

      setTimeout(() => {
        onSuccess();
        onOpenChange(false);
        reset();
      }, 100);
    } catch (error) {
      setError(
        error instanceof Error ? error.message : 'Failed to change password'
      );
    } finally {
      setIsSubmitting(false);
    }
  };

  const handleClose = () => {
    onOpenChange(false);
    reset();
    setError(null);
  };

  return (
    <Dialog open={open} onOpenChange={handleClose}>
      <DialogContent className='sm:max-w-[425px]'>
        <DialogHeader>
          <DialogTitle>Change Password</DialogTitle>
        </DialogHeader>

        <form onSubmit={handleSubmit(onSubmit)} className='space-y-4'>
          {error && <ErrorMessage customMessage={error} />}

          <InputController
            name='currentPassword'
            label='Current Password'
            control={control}
            type='password'
            placeholder='Enter current password'
            disabled={isSubmitting}
            showPasswordToggle
          />

          <InputController
            name='newPassword'
            label='New Password'
            control={control}
            type='password'
            placeholder='Enter new password'
            disabled={isSubmitting}
            showPasswordToggle
          />

          <InputController
            name='confirmPassword'
            label='Confirm New Password'
            control={control}
            type='password'
            placeholder='Confirm new password'
            disabled={isSubmitting}
            showPasswordToggle
          />

          <DialogFooter>
            <Button
              type='button'
              variant='secondary'
              onClick={handleClose}
              disabled={isSubmitting}
            >
              Cancel
            </Button>
            <Button type='submit' variant='primary' disabled={isSubmitting}>
              {isSubmitting ? 'Changing...' : 'Change Password'}
            </Button>
          </DialogFooter>
        </form>
      </DialogContent>
    </Dialog>
  );
}

export default ChangePasswordModal;
