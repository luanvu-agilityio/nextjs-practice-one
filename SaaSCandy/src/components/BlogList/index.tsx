import { getAllPosts } from '@/helpers';
import { BlogCard } from '../BlogCard';

async function BlogList() {
  const posts = await getAllPosts();
  return (
    <>
      {posts.map(p => (
        <BlogCard key={p.slug} {...p} />
      ))}
    </>
  );
}

export default BlogList;
