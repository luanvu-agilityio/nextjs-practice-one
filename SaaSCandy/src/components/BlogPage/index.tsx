// Components
import Section from '../common/Section';
import { BlogCard } from '../BlogCard';

// Helper
import { getAllPosts } from '@/helpers';

async function BlogPage() {
  const posts = await getAllPosts();

  return (
    <Section className='bg-white'>
      <div className='grid sm:grid-cols-2 lg:grid-cols-3 gap-6'>
        {posts.map(p => (
          <BlogCard key={p.slug} {...p} />
        ))}
      </div>
    </Section>
  );
}
export default BlogPage;
