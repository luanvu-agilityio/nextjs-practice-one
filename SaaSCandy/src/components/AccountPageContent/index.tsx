'use client';

import { Fragment, useEffect, useState } from 'react';
import Link from 'next/link';
import { useSession, signOut } from '@/lib/auth-client';
import { useRouter } from 'next/navigation';
import Image from 'next/image';

// Components
import { Button, Heading, Typography } from '@/components/common';
import {
  Breadcrumb,
  BreadcrumbList,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbSeparator,
  BreadcrumbPage,
} from '@/components/common/Breadcrumb';
import { Spinner } from '@/components';
import { showToast } from '@/components/common/Toast';
import EditProfileModal from '../EditProfileModal';
import ChangePasswordModal from '../ChangePasswordModal';

// Icons
import LogoIcon from '@/components/icons/Logo';

// Utils & Constants
import { ROUTES, TOAST_MESSAGES } from '@/constants';

// Types
import { TOAST_VARIANTS } from '@/types/toast';

function AccountPageContent() {
  const { data: session, isPending } = useSession();
  const router = useRouter();
  const [isEditProfileOpen, setIsEditProfileOpen] = useState(false);
  const [isChangePasswordOpen, setIsChangePasswordOpen] = useState(false);

  // Redirect if not authenticated
  useEffect(() => {
    if (!isPending && !session?.user) {
      router.push(ROUTES.SIGN_IN);
    }
  }, [router, session?.user, isPending]);

  if (isPending) {
    return (
      <div className='flex justify-center items-center min-h-screen'>
        <Spinner />
      </div>
    );
  }

  if (!session?.user) {
    return null;
  }

  const {
    user: { name, email, image },
  } = session;

  const handleGoBack = () => {
    router.back();
  };

  const handleSignOut = async () => {
    try {
      await signOut();
      router.push(ROUTES.HOME);
      router.refresh();
      showToast({
        ...TOAST_MESSAGES.ACCOUNT.SIGNOUT_SUCCESS,
        variant: TOAST_VARIANTS.SUCCESS,
      });
    } catch (error) {
      console.error('Sign out error:', error);
      showToast({
        ...TOAST_MESSAGES.ACCOUNT.SIGNOUT_ERROR,
        variant: TOAST_VARIANTS.ERROR,
      });
    }
  };

  const handleEditProfileSuccess = () => {
    showToast({
      ...TOAST_MESSAGES.ACCOUNT.PROFILE_UPDATE_SUCCESS,
      variant: TOAST_VARIANTS.SUCCESS,
    });
  };

  const handleChangePasswordSuccess = () => {
    showToast({
      ...TOAST_MESSAGES.ACCOUNT.PASSWORD_CHANGE_SUCCESS,
      variant: TOAST_VARIANTS.SUCCESS,
    });
  };

  const breadcrumbs = [
    { label: 'Home', href: ROUTES.HOME, isActive: false },
    { label: 'Account', href: ROUTES.ACCOUNT, isActive: true },
  ];

  const isSocialAccount = !!image;

  return (
    <>
      <div className='flex flex-col items-center max-w-[1296px] mx-auto min-h-screen py-4 sm:py-8 px-4 sm:px-6'>
        {/* Header Section  */}
        <div className='pb-6 sm:pb-12 w-full'>
          <Heading
            as='h1'
            size='xl'
            content='Account Details'
            className='text-center text-2xl sm:text-3xl lg:text-4xl'
          />

          {/* Breadcrumb */}
          <Breadcrumb className='justify-center mt-2 sm:mt-4'>
            <BreadcrumbList>
              {breadcrumbs.map((item, index) => (
                <Fragment key={item.label}>
                  <BreadcrumbItem>
                    {item.isActive ? (
                      <BreadcrumbPage className='text-sm sm:text-xs'>
                        {item.label}
                      </BreadcrumbPage>
                    ) : (
                      <BreadcrumbLink
                        href={item.href}
                        className='text-sm sm:text-xs'
                      >
                        {item.label}
                      </BreadcrumbLink>
                    )}
                  </BreadcrumbItem>
                  {index < breadcrumbs.length - 1 && <BreadcrumbSeparator />}
                </Fragment>
              ))}
            </BreadcrumbList>
          </Breadcrumb>
        </div>

        {/* Account Detail Section  */}
        <div className='py-8 px-4 sm:py-14 sm:px-16 w-full lg:max-w-159 rounded-2xl sm:rounded-4xl border border-form-border-color shadow-form bg-white'>
          <div className='flex flex-col gap-6 sm:gap-10'>
            {/* Logo */}
            <div className='flex justify-center'>
              <Link
                href={ROUTES.HOME}
                className='flex items-center gap-2 sm:gap-3'
              >
                <LogoIcon className='w-8 h-8 sm:w-10 sm:h-10' />
                <span className='text-2xl sm:text-3xl font-secondary text-primary'>
                  SaaS<span className='font-medium'>Candy</span>
                </span>
              </Link>
            </div>

            {/* User Information */}
            <div className='space-y-6 sm:space-y-8'>
              {/* Profile Header  */}
              <div className='text-center space-y-3 sm:space-y-4'>
                {image ? (
                  <div className='flex justify-center'>
                    <Image
                      src={image}
                      alt='Profile picture'
                      width={96}
                      height={96}
                      className='w-20 h-20 sm:w-24 sm:h-24 rounded-full border-4 border-orange-background object-cover'
                    />
                  </div>
                ) : (
                  <div className='flex justify-center'>
                    <div className='w-20 h-20 sm:w-24 sm:h-24 rounded-full bg-orange-background flex items-center justify-center text-white text-xl sm:text-2xl font-bold'>
                      {name?.charAt(0) || email?.charAt(0) || 'U'}
                    </div>
                  </div>
                )}

                <div>
                  <Heading
                    as='h2'
                    size='md'
                    className='text-primary text-xl sm:text-2xl'
                    content={name || 'User'}
                  />

                  <Typography className='text-gray-background mt-1 text-sm sm:text-xs break-all'>
                    {email}
                  </Typography>
                </div>
              </div>

              {/* Account Information Cards */}
              <div className='space-y-3 sm:space-y-4'>
                {/* Personal Information */}
                <div className='bg-blue-foreground opacity-80 rounded-lg p-4 sm:p-6'>
                  <Heading
                    as='h3'
                    size='md'
                    className='text-primary mb-3 sm:mb-4 text-lg sm:text-xl'
                    content='Personal Information'
                  />

                  <div className='space-y-2 sm:space-y-3'>
                    <div className='flex flex-col sm:flex-row sm:justify-between sm:items-center gap-1 sm:gap-0  sm:text-lg'>
                      <span className='text-blue-background font-bold'>
                        Name:
                      </span>
                      <span className='font-medium text-primary'>
                        {name || 'Not provided'}
                      </span>
                    </div>
                    <div className='flex flex-col sm:flex-row sm:justify-between sm:items-center gap-1 sm:gap-0  sm:text-lg'>
                      <span className='text-blue-background font-bold'>
                        Email:
                      </span>
                      <span className='font-medium text-primary break-all'>
                        {email}
                      </span>
                    </div>
                    <div className='flex flex-col sm:flex-row sm:justify-between sm:items-center gap-1 sm:gap-0 sm:text-lg'>
                      <span className='text-blue-background font-bold'>
                        Account Type:
                      </span>
                      <span className='px-2 py-1 bg-blue-foreground text-blue-background rounded-full text-sm sm:text-md inline-block w-fit'>
                        {isSocialAccount ? 'Social Account' : 'Email Account'}
                      </span>
                    </div>
                  </div>
                </div>

                {/* Account Settings */}
                <div className='bg-blue-foreground opacity-80 rounded-lg p-4 sm:p-6'>
                  <Heading
                    as='h3'
                    size='md'
                    className='text-primary mb-3 sm:mb-4 text-lg sm:text-xl'
                    content='Account Settings'
                  />

                  <div className='space-y-2 sm:space-y-3'>
                    <Button
                      variant='secondary'
                      className='w-full justify-start font-semibold sm:text-lg'
                      onClick={() => setIsEditProfileOpen(true)}
                    >
                      Edit Profile
                    </Button>

                    <Button
                      variant='secondary'
                      className='w-full justify-start font-semibold  sm:text-lg'
                      onClick={() => setIsChangePasswordOpen(true)}
                      disabled={isSocialAccount}
                    >
                      <span className='block sm:inline'>Change Password</span>
                      {isSocialAccount && (
                        <span className='block sm:inline text-xs sm:text-sm ml-0 sm:ml-1 mt-1 sm:mt-0'>
                          (Not available for social accounts)
                        </span>
                      )}
                    </Button>

                    <Button
                      variant='secondary'
                      className='w-full justify-start font-semibold  sm:text-lg'
                      disabled
                    >
                      Privacy Settings (Coming Soon)
                    </Button>
                  </div>
                </div>
              </div>

              {/* Action Buttons  */}
              <div className='flex flex-col sm:flex-row gap-3 sm:gap-4 pt-4 sm:pt-6'>
                <Button
                  variant='secondary'
                  className='flex-1 w-full  sm:text-lg'
                  onClick={handleGoBack}
                >
                  Back
                </Button>
                <Button
                  variant='tertiary'
                  className='flex-1 w-full  sm:text-lg'
                  onClick={handleSignOut}
                >
                  Sign Out
                </Button>
              </div>
            </div>
          </div>
        </div>
      </div>

      {/* Modals */}
      <EditProfileModal
        open={isEditProfileOpen}
        onOpenChange={setIsEditProfileOpen}
        onSuccess={handleEditProfileSuccess}
      />

      <ChangePasswordModal
        open={isChangePasswordOpen}
        onOpenChange={setIsChangePasswordOpen}
        onSuccess={handleChangePasswordSuccess}
      />
    </>
  );
}

export default AccountPageContent;
