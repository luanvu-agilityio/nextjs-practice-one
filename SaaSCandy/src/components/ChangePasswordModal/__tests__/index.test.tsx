import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import ChangePasswordModal from '../index';
import { changePassword } from '@/service';
import { showToast } from '@/components/common';

jest.mock('@/lib/auth-client', () => ({
  useSession: () => ({
    data: { user: { id: 'test-user-id' } },
  }),
}));

jest.mock('@/service');
jest.mock('@/components/common', () => ({
  Button: ({
    children,
    ...props
  }: { children: React.ReactNode } & Record<string, unknown>) => (
    <button {...props}>{children}</button>
  ),
  showToast: jest.fn(),
}));

jest.mock('@/components/common/ErrorMessage', () => ({
  ErrorMessage: ({ customMessage }: { customMessage: string }) => (
    <div>{customMessage}</div>
  ),
}));

jest.mock('@/components/common/InputController', () => ({
  InputController: ({ name, label }: { name: string; label: string }) => (
    <input data-testid={name} placeholder={label} />
  ),
}));

jest.mock('@/components', () => ({
  Dialog: ({
    children,
    open,
  }: { children: React.ReactNode } & Record<string, unknown>) =>
    open ? <div>{children}</div> : null,
  DialogContent: ({
    children,
  }: { children: React.ReactNode } & Record<string, unknown>) => (
    <div>{children}</div>
  ),
  DialogHeader: ({
    children,
  }: { children: React.ReactNode } & Record<string, unknown>) => (
    <div>{children}</div>
  ),
  DialogTitle: ({
    children,
  }: { children: React.ReactNode } & Record<string, unknown>) => (
    <h2>{children}</h2>
  ),
  DialogFooter: ({
    children,
  }: { children: React.ReactNode } & Record<string, unknown>) => (
    <div>{children}</div>
  ),
}));

describe('ChangePasswordModal', () => {
  const mockOnOpenChange = jest.fn();
  const mockOnSuccess = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when open', () => {
    const { container } = render(
      <ChangePasswordModal
        open={true}
        onOpenChange={mockOnOpenChange}
        onSuccess={mockOnSuccess}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
