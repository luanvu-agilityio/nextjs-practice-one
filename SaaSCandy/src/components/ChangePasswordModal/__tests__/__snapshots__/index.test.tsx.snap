// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ChangePasswordModal renders correctly when open 1`] = `
<div>
  <div>
    <div>
      <div>
        <h2>
          Change Password
        </h2>
      </div>
      <form
        class="space-y-4"
      >
        <input
          data-testid="currentPassword"
          placeholder="Current Password"
        />
        <input
          data-testid="newPassword"
          placeholder="New Password"
        />
        <input
          data-testid="confirmPassword"
          placeholder="Confirm New Password"
        />
        <div>
          <button
            type="button"
            variant="secondary"
          >
            Cancel
          </button>
          <button
            type="submit"
            variant="primary"
          >
            Change Password
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
